src/utils/error_utils.go

```diff
@@ -0,0 +1,45 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/go-openapi/swag"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+)
+
+type AssistedServiceError interface {
+	Error() string
+	GetPayload() *models.Error
+}
+
+type AssistedServiceInfraError interface {
+	Error() string
+	GetPayload() *models.InfraError
+}
+
+func BuildMessageFromAssistedServiceError(err error) string {
+	switch err := err.(type) {
+	case AssistedServiceError:
+		payload := err.GetPayload()
+		return fmt.Sprintf(
+			"AssistedServiceError Code: %s Href: %s ID: %d Kind: %s Reason: %s",
+			swag.StringValue(payload.Code),
+			swag.StringValue(payload.Href),
+			swag.Int32Value(payload.ID),
+			swag.StringValue(payload.Kind),
+			swag.StringValue(payload.Reason))
+	case AssistedServiceInfraError:
```

Comment:
Curious. what is AssistedServiceInfraError?

Reply:
A hack for Auth/Authz the infra return `runtime.APIError` we have a way to fix it but didn't handled it yet.